addons/website/static/src/js/menu/edit.js

```diff
@@ -39,29 +41,28 @@ var EditPageMenu = websiteNavbarData.WebsiteNavbarActionWidget.extend({
     start: function () {
         var def = this._super.apply(this, arguments);
 
-        // If we auto start the editor, do not show a welcome message
-        if (this._editorAutoStart) {
-            this._startEditMode();
-            return def;
-        }
-
         // Check that the page is empty
-        var $wrap = $('#wrapwrap.homepage #wrap'); // TODO find this element another way
-        if (!$wrap.length || $wrap.html().trim() !== '') {
-            return def;
-        }
+        var $wrap = this._targetForEdition().find('#wrap');
+        this.$wrap = $wrap;
 
-        // If readonly empty page, show the welcome message
-        this.$welcomeMessage = $(core.qweb.render('website.homepage_editor_welcome_message'));
-        this.$welcomeMessage.css('min-height', $wrap.parent('main').height() - ($wrap.outerHeight(true) - $wrap.height()));
-        $wrap.empty().append(this.$welcomeMessage);
+        if ($wrap.length && $wrap.html().trim() === '') {
+            // If readonly empty page, show the welcome message
+            this.$welcomeMessage = $(core.qweb.render('website.homepage_editor_welcome_message'));
+            this.$welcomeMessage.addClass('o_homepage_editor_welcome_message');
+            this.$welcomeMessage.css('min-height', $wrap.parent('main').height() - ($wrap.outerHeight(true) - $wrap.height()));
+            $wrap.empty().append(this.$welcomeMessage);
+        }
 
         setTimeout(function () {
             if ($('.o_tooltip.o_animated').length) {
                 $('.o_tooltip_container').addClass('show');
             }
         }, 1000); // ugly hack to wait that tooltip is loaded
 
+        // If we auto start the editor, do not show a welcome message
+        if (this._editorAutoStart) {
+            return this._startEditMode();
```

Comment:
in this case, we do not wait for the `def` of super, is this normal ?

Reply:
@Gorash